t/03-testapi.t

```diff
@@ -683,4 +683,145 @@ subtest 'autoinst_url' => sub {
+            },
+            {
+                cmd => 'backend_mouse_set',
+                x   => 200,
+                y   => 200
+            },
+            {
+                bstate => 0,
+                button => 'left',
+                cmd    => 'backend_mouse_button'
+            },
+    ], 'mouse drag (startpoint using needle) was succesful') or diag explain $cmds;
+    
+    # Startpoint from coordinates, endpoint from a needle.
+    $cmds = [];
+    mouse_drag(endpoint => 'foo', startx => $startx, starty => $starty, button => $button, timeout => 30);
+    is_deeply($cmds, [
+            {
+                cmd => 'backend_mouse_set',
+                x   => 0,
+                y   => 0
+            },
+            {
+                bstate => 1,
+                button => 'left',
+                cmd    => 'backend_mouse_button'
+            },
+            {
+                cmd => 'backend_mouse_set',
+                x   => 110,
+                y   => 110
+            },
+            {
+                bstate => 0,
+                button => 'left',
+                cmd    => 'backend_mouse_button'
+            },
+    ], 'mouse drag (endpoint using needle) was succesful') or diag explain $cmds;
+    
+    # Using coordinates only.
+    $cmds = [];
+    mouse_drag(endx => $endx, endy => $endy, startx => $startx, starty => $starty, button => $button, timeout => 30);
+    is_deeply($cmds, [
+            {
+                cmd => 'backend_mouse_set',
+                x   => 0,
+                y   => 0
+            },
+            {
+                bstate => 1,
+                button => 'left',
+                cmd    => 'backend_mouse_button'
+            },
+            {
+                cmd => 'backend_mouse_set',
+                x   => 200,
+                y   => 200
+            },
+            {
+                bstate => 0,
+                button => 'left',
+                cmd    => 'backend_mouse_button'
+            },
+    ], 'mouse drag (using coordinates only) was succesful') or diag explain $cmds;
+    
+    # Both needle and coordinates provided for startpoint (coordinates should win).
+    $cmds = [];
+    mouse_drag(startpoint => "foo", endx => $endx, endy => $endy, startx => $startx, starty => $starty, button => $button, timeout => 30);
+    is_deeply($cmds, [
+            {
+                cmd => 'backend_mouse_set',
+                x   => 0,
+                y   => 0
+            },
+            {
+                bstate => 1,
+                button => 'left',
+                cmd    => 'backend_mouse_button'
+            },
+            {
+                cmd => 'backend_mouse_set',
+                x   => 200,
+                y   => 200
+            },
+            {
+                bstate => 0,
+                button => 'left',
+                cmd    => 'backend_mouse_button'
+            },
+    ], 'mouse drag (using coordinates only) was succesful') or diag explain $cmds;
```

Comment:
I guess successful is written differently but I'd actually remove it completely because that's kind of implied for a test description.

Reply:
All right.